sql/src/main/java/io/crate/execution/engine/indexing/ColumnIndexWriterProjector.java

```diff
@@ -94,20 +96,31 @@ public ColumnIndexWriterProjector(ClusterService clusterService,
             updateColumnNames = convert.targetNames();
             assignments = convert.sources();
         }
+
+        Symbol[] returnValueOrNull = returnValues.isEmpty() ? null : returnValues.toArray(new Symbol[]{});
```

Comment:
very minor:  Isn't `toArray(new Object[0])` still faster, safer than `toArray(new Object[])`? (see https://shipilev.net/blog/2016/arrays-wisdom-ancients/)

Reply:
Good point i was not aware of this. (btw, great link)